libafl/src/events/mod.rs

```diff
@@ -76,6 +86,23 @@ impl Handler for ShutdownSignalData {
         _info: &mut siginfo_t,
         _context: Option<&mut ucontext_t>,
     ) {
+        println!(
+            "in handler! {} {}",
+            self.shmem_allocated,
+            std::process::id()
+        );
+        // if this process has not allocated any shmem. then simply exit()
+        if !self.shmem_allocated {
+            unsafe {
+                #[cfg(unix)]
+                libc::_exit(0);
+
+                #[cfg(windows)]
+                windows::Win32::System::Threading::ExitProcess(1);
+            }
+        }
+
+        // else wait till the next is_shutting_down() is called. then the process will exit throught main().
```

Comment:
Why not always exit through main? What does exiting have to do with `shmem_allocated`? Seems not clear to me

Reply:
Maybe consider a different/clearer variable name

Reply:
because they can't do so with the current design.

This function, launch() is usually called from libafl_main()
Usually if you are still in this launch() function waiting for the child process to finish, 
it is easy for this guy to exit through main if you exit this function from this line 
https://github.com/AFLplusplus/LibAFL/blob/main/libafl/src/events/simple.rs#L520

but for other processes like child process that has just received sigint during, say, inprocess::run_target. this process does not know how to return back to the libafl_main().

Reply:
> What does exiting have to do with shmem_allocated?

if you exit from main() then you can call the drop() for the shmem you allocated.
otherwise if you just exited with libc::exit(0) or any syscalls these resource are not cleaned.
so "exit from main" = "clean up resources used"

Reply:
renamed